src/models/context/AccountContextManager.ts

```diff
@@ -38,16 +38,16 @@ export class AccountContextManager {
                 this._currentKubeconfigPath = kubeConfigPathResult.hostPath;
                 if (this._fsWatcher == null) {
                     this._fsWatcher = FileSystemWatcher.watch(this._currentKubeconfigPath);
-                    this._fsWatcher.on(`change`, path => {
-                        this._logger.trace(TelemetryEvent.AccountContextManager_KubeconfigChange);
-                        this._kubeconfigChanged.trigger();
-                    });
                 }
                 else {
                     this._fsWatcher.unwatch(oldKubeconfigPath);
                     this._fsWatcher.add(this._currentKubeconfigPath);
                 }
             }
+            // The kubeconfig file is watched for changes, and the event is triggered when the file is changed.
+            this._fsWatcher.on(`change`, (path) => {
```

Comment:
💡 this is interesting hence asking, moving this out means it will skip the `if clause` and what was the pusepose of that if clause? 

Previously it was this:

So thinking outlaud as to what was the reason and extent the change was trigger impact :) no clue, in case you know what is the reason behind this. aka why previously it was cushioned with `if` clause previously?

```*.ts
         if (this._fsWatcher == null) {
                    this._fsWatcher = FileSystemWatcher.watch(this._currentKubeconfigPath);
                    this._fsWatcher.on(`change`, path => {
                        this._logger.trace(TelemetryEvent.AccountContextManager_KubeconfigChange);
                        this._kubeconfigChanged.trigger();
                    });
                }
                else {
                    this._fsWatcher.unwatch(oldKubeconfigPath);
                    this._fsWatcher.add(this._currentKubeconfigPath);
                }
```

Reply:
Good question @Tatsinnit I was also surprised how this was logging 20M times being inside if condition. According to the library this code will execute whenever there is change event triggered on the directory or file path it's watching so if condition doesn't make sense and caused confusion here. so moved the code below for more readability,

Reply:
@Tatsinnit are we good with this ? can we resolve this and merge the PR ?